src/header/common/strict_transport_security.rs

```diff
@@ -0,0 +1,259 @@
+use header::{Header, HeaderFormat};
+use std::fmt::{self};
+
+/// `StrictTransportSecurity` header, defined in [RFC6797](https://tools.ietf.org/html/rfc6797)
+///
+/// This specification defines a mechanism enabling web sites to declare
+/// themselves accessible only via secure connections and/or for users to be
+/// able to direct their user agent(s) to interact with given sites only over
+/// secure connections.  This overall policy is referred to as HTTP Strict
+/// Transport Security (HSTS).  The policy is declared by web sites via the
+/// Strict-Transport-Security HTTP response header field and/or by other means,
+/// such as user agent configuration, for example.
+///
+/// # Example values
+/// * `max-age=31536000`
+/// * `max-age=15768000 ; includeSubDomains`
+/// 
+/// # Example
+/// ```
+/// # extern crate hyper;
+/// # fn main() {
+/// use hyper::header::{Headers, StrictTransportSecurity};
+///
+/// let mut headers = Headers::new();
+///
+/// headers.set(
+///    StrictTransportSecurity(true, 31536000u64)
+/// );
+/// # }
+/// ```
+#[derive(Clone, PartialEq, Debug)]
+pub struct StrictTransportSecurity(pub bool, pub u64);
```

Comment:
It seems slightly odd that the header outputs with the boolean at the end, but the struct has it first.

Also, booleans have a habit of being confusing without the explicit name right next to it. How about instead:

```rust
pub struct StrictTransportSecurity {
    pub max_age: u64,
    pub includes_subdomains: bool,
}
```

Additionally, and I'm not sure if the value this might provide carries its own weight, but maybe `StrictTransportSecurity::including_subdomain(1_000_000)` and `StrictTransportSecurity::excluding_subdomain(1_000_000)` constructors might be useful?